electron_app/helpers/handleDisk.js

```diff
@@ -1,18 +1,49 @@
 const fs = require('fs');
 const mkdirp = require('mkdirp');
+const diskspace = require('diskspace');
+
+let saveConditions = { enoughSpace: Boolean, allowed: Boolean };
+let Conditions = {error: false, errorMessage: ''};
 
 const handleFolder = (directory) => {
-  return new Promise((resolve, reject) => {
-    if (!fs.existsSync(directory)) {
+    return new Promise((resolve, reject) => {
+        if (!fs.existsSync(directory)) {
+
+            mkdirp.sync(directory, (err) => {
+                if (err) {
+                    reject(err);
+                }
+            });
+        }
+        resolve(directory);
+    });
+};
 
-      mkdirp.sync(directory, (err) => {
-        if (err) {
-          reject(err);
+const handleDiskSpace = (destination, totalFileSize) => {
+    var drive = destination;
+    Conditions = {error: Boolean, errorMessage: String};    
+    return new Promise((res, rej) => {
+        if (process.platform == 'win32') {
```

Comment:
This should use `===`, not `==`, correct? I thought we had a lint for this; but now see that this is a regular .js file not a .ts file. Can this code be written as .ts as well? I think that would be better, for readability and maintenance.

Reply:
I've corrected `===` operator. 
On the other hand, what you mentioned regarding `.js` to `.ts` files sounds reasonable, so yes, please lets create a Jira ticket to address this improvement.